htdocs/product/class/api_products.class.php

```diff
@@ -186,6 +187,11 @@ public function index($sortfield = "t.ref", $sortorder = 'ASC', $limit = 100, $p
 			$sql .= ", ".MAIN_DB_PREFIX."categorie_product as c";
 		}
 		$sql .= ' WHERE t.entity IN ('.getEntity('product').')';
+
+		if ($without_variants) {
+			$sql .= ' AND t.rowid in (select distinct fk_product_parent from '.MAIN_DB_PREFIX.'product_attribute_combination)';
```

Comment:
Some products are A) common products, some are B) parent of variant and some are C) variant.
Here adding the test will return all products B (right).
Why not but it means label of parameter is wrong (without_variant means A for me).
If what you want if really B, i suggest to rename parameter into "$parent_of_variant"

Or better, rename it into "$variant_filter" with 2 values (0 = all, 1=A only, 2=B only, 3=C only). Need more change. First suggestion correction will be enough.

Reply:
Good point. I will make the change and push it again.